mtz-wizard.html

```diff
@@ -147,12 +148,21 @@
        * Runs the validation logic on all steps with a form bound to them.
        *
        * @param {HTMLElement[]} [steps = this.steps]
-       * @return {Boolean}
+       * @return {Promise} - resolves with a Boolean that says if the wizard is valid or not
        */
       validate(steps = this.steps) {
-        return (steps || []).reduce((prev, step) => {
-          return prev && (step.validate ? step.validate() : true);
-        }, true);
+        return Promise.all((steps || []).
+          map((step) => {
+            return step.validate ?
```

Comment:
And yet still my comment stands.

>  Promise.resolve(!step.validate || step.validate())? This would allow us to handle non-promise validation as well.